crates/re_data_store/src/store_db.rs

```diff
@@ -36,12 +37,40 @@ impl Default for EntityDb {
             entity_path_from_hash: Default::default(),
             times_per_timeline: Default::default(),
             tree: crate::EntityTree::root(),
-            data_store: re_arrow_store::DataStore::new(
-                InstanceKey::name(),
-                DataStoreConfig::default(),
-            ),
+            data_store: DataStore::new(InstanceKey::name(), DataStoreConfig::default()),
+        }
+    }
+}
+
+/// See [`insert_row_with_retries`].
+const NUM_INSERT_ROW_ATTEMPTS: usize = 100;
+
+/// Inserts a [`DataRow`] into the [`DataStore`].
+///
+/// Retries a maximum of `num_attempts` times if the row couldn't be inserted because of a
+/// duplicated [`RowId`], rerolling the [`RowId`] as needed between attempts.
+///
+/// Returns the actual [`DataRow`] that was successfully inserted, if any.
+///
+/// The default value of `num_attempts` (see [`NUM_INSERT_ROW_ATTEMPTS`]) should be (way) more than
+/// enough for all valid use cases.
+fn insert_row_with_retries(
+    store: &mut DataStore,
+    mut row: DataRow,
+    num_attempts: usize,
+) -> re_arrow_store::WriteResult<DataRow> {
+    for _ in 0..num_attempts {
+        match store.insert_row(&row) {
+            Ok(_) => return Ok(row),
+            Err(re_arrow_store::WriteError::ReusedRowId(_)) => {
+                warn_once!("Found duplicated RowId({})", row.row_id());
```

Comment:
Shouldn't even be `warn_once` here, this is expected to happen once every blue moon..